GAE/src/org/waterforpeople/mapping/dataexport/GraphicalSurveySummaryExporter.java

```diff
@@ -1646,18 +1637,54 @@ private void writeSummaryReport(
                             anchor.setRow2(tableTopRow + CHART_CELL_HEIGHT);
                             anchor.setAnchorType(2);
                             patriarch.createPicture(anchor, indx);
-                            if (tableTopRow + CHART_CELL_HEIGHT > tableBottomRow) {
-                                curRow = tableTopRow + CHART_CELL_HEIGHT;
+                            if (tableTopRow + CHART_CELL_HEIGHT > bottomRow) {
+                                bottomRow = tableTopRow + CHART_CELL_HEIGHT;
+                            }
+                        }
+                    }
+
+                    if (doDataTable) {
+                        curRow = tableTopRow;
+                        //header
+                        row = getRow(curRow++, sheet);
+                        createCell(row, 1, FREQ_LABEL, headerStyle);
+                        createCell(row, 2, PCT_LABEL, headerStyle);
+                        
+                        //items
+                        for (int i=0; i<labels.size(); i++) {
+                            row = getRow(curRow++, sheet);
+                            createCell(row, 0, labels.get(i));
+                            createCell(row, 1, values.get(i));
+                            if (sampleTotal > 0) {
+                                createCell(row, 2,
+                                        PCT_FMT.format((Double.parseDouble(values.get(i))
+                                                / sampleTotal)));
+                            } else {
+                                createCell(row, 2, PCT_FMT.format(0));
                             }
+                            if (i % 50 == 49) {
+                                //flush to stay within window
+                                ((SXSSFSheet) sheet).flushRows(0);
+                            }
+                        }
+                        
+                        //total
+                        row = getRow(curRow++, sheet);
+                        createCell(row, 0, TOTAL_LABEL);
+                        createCell(row, 1, sampleTotal + "");
+                        if (curRow > bottomRow) {
+                            bottomRow = curRow;
                         }
                     }
+                    curRow = bottomRow;
 
                     // add a blank row between questions
                     getRow(curRow++, sheet);
                     // flush the sheet so far to disk; we will not go back up
+                    // This is crucial, not just an optimisation!
+                    // File broken if we have too many lines between each flush
                     ((SXSSFSheet) sheet).flushRows(0); // retain 0 last rows and
```

Comment:
According to the docs the flushing should happen automatically when accessing `createRow()`.  What i am not sure about is whether the docs are for the more recent version or still valid for the 3.8 that we are using

Reply:
Probably it is just the going back up that is crucial. I can do the experiment.

Reply:
Yes, holds for 3.8 also